### PR TITLE
mpu6050: driver optimize

### PR DESCRIPTION
**Pull request by joy4eg**
- Optimize sysfs interface
- Bugfix

### CODE REVIEW

**Comment by an1kh** [comment]
Great idea about sysfs_create_group!

File: mpu6050/mpu6050.c [hunk 1]
```diff
@@ -272,4 +155,4 @@ module_exit(mpu6050_exit);
 MODULE_AUTHOR("Andriy.Khulap <andriy.khulap@globallogic.com>");
```
Comment (an1kh) [inline mpu6050/mpu6050.c]: Please set your name here

**Comment by an1kh** [comment]
As I understand, sysfs group will be created in /sys/class/i2c-dev/i2c-2/device/2-0068/
Is this correct?

**Comment by joy4eg** [comment]
>As I understand, sysfs group will be created in /sys/class/i2c-dev/i2c-2/device/2-0068/
Is this correct?

@an1kh Yep, it's correct.

**Comment by an1kh** [comment]
> Yep, it's correct.

I investigated your method further and found that it can be simplified even more by pointing to attribute groups in driver member of i2c_driver:
```
struct class_attribute attr_a = {
	.attr = { .name = "attr_a", .mode = S_IRUGO },
	.show = attr_multi_show,
};

struct class_attribute attr_b = {
	.attr = { .name = "attr_b", .mode = S_IRUGO },
	.show = attr_multi_show,
};

struct class_attribute attr_c = {
	.attr = { .name = "attr_c", .mode = S_IRUGO },
	.show = attr_multi_show,
};

static struct attribute *mpu6050_attrs_group_1[] = {
	(struct attribute *)&attr_a,
	(struct attribute *)&attr_b,
	NULL
};

static struct attribute *mpu6050_attrs_group_2[] = {
	(struct attribute *)&attr_b,
	(struct attribute *)&attr_c,
	NULL
};

static struct attribute_group mpu6050_attr_group_1 = {
	.name = "attr_group_1",
	.attrs = mpu6050_attrs_group_1,
};

static struct attribute_group mpu6050_attr_group_2 = {
	.name = "attr_group_2",
	.attrs = mpu6050_attrs_group_2,
};


static const struct attribute_group *mpu6050_attr_groups[] = {
	&mpu6050_attr_group_1,
	&mpu6050_attr_group_2,
	NULL
};

static const struct i2c_device_id mpu6050_idtable [] = {
	{ "mpu6050", 0 },
	{ }
};

MODULE_DEVICE_TABLE(i2c, mpu6050_idtable);

static struct i2c_driver mpu6050_i2c_driver = {
	.driver = {
		.name = "gl_mpu6050",
		.groups = mpu6050_attr_groups,
	},

	.probe = mpu6050_probe,
	.remove = mpu6050_remove,
	.id_table = mpu6050_idtable,
};
```
But anyway, this lead to different paths in case of i2c bus and device address variations. So application should implement some search mechanism instead of hard-coded path.

**Comment by joy4eg** [comment]
@an1kh I would prefer to leave current version "as is", because with `struct class_attribute` we cannot pass additional value to our handlers (register number) and current implementation is more compact :)

>But anyway, this lead to different paths in case of i2c bus and device address variations.

For example, we can create a compatibility link for our device using `class_compat_register`/`class_compat_create_link` and read values from `/sys/class/mpu6050/2-0068/<param>`

**Comment by an1kh** [comment]
@joy4eg, its up to you, all ok as is.
As I understood, automatic creation is suggested by http://kroah.com/log/blog/2013/06/26/how-to-create-a-sysfs-file-correctly/

**Comment by joy4eg** [comment]
@an1kh @AleksandrBulyshchenko How about to merge this PR ? :)

**Comment by an1kh** [comment]
I have no objections.

**Review by an1kh — APPROVED** [review]

File: mpu6050/mpu6050.c [hunk 2]
```diff
@@ -4,70 +4,87 @@
...
-	int gyro_values[3];
-	int temperature;
-};
+	struct dev_ext_attribute *ea = container_of(attr, struct dev_ext_attribute, attr);
+	struct i2c_client *drv_client = to_i2c_client(dev);
+	int val;
 
-static struct mpu6050_data g_mpu6050_data;
+	val = (s16)((u16)i2c_smbus_read_word_swapped(drv_client, (u32)ea->var));
+	val = (val + 12420 + 170) / 340;
```
Comment (AleksandrBulyshchenko) [inline mpu6050/mpu6050.c]: Try to avoid magic numbers

**Review by AleksandrBulyshchenko — COMMENTED** [review]
@joy4eg,
I like the solution overall. :+1:

Only few comments from my side:
- commit messages should be more descriptive, as well as PR description;
- while your branch hasn't been merged, prefer to update your commits with cleanups and fixes (use interactive rebase), rather than provide commit them separately;
- don't mix your own changes with commits merged from master - they can be pushed directly, without PR;

Please follow [wiki](https://github.com/Kernel-GL-HRK/gl-kernel-training-2017/wiki/Pull-Request-Guideline) suggestions when creating PRs.

**Pull request merged**